docs/concepts/custom-ui-headless-api.mdx

```diff
@@ -0,0 +1,662 @@
+different programming languages.
+
+This documents' focus is to provide concrete examples on custom UI integration. Flows are the fundamental of Ory's self-service
+APIs and are used to accomplish a variety of goals, such as `login`, `registration`, `settings`, `verification` and `recovery`.
+
+The self-service `flows` are standardized so that it follows the same steps regardless of which `flow` is integrated.
+
+A general overview of the flow can be summed up in two steps:
+
+1. Get or create the `flow` data, e.g. login, registration etc.
+2. Submit the `flow` with user data (e.g. username and password)
+
+For more information on the concepts, take a look at the [Self-service flow](../kratos/self-service) documentation.
+
+## Creating a flow
+
+All browser flows can be initialized with the optional `?return_to=<url>` query parameter. This parameter is used to redirect the
+user to the specified URL after the flow is successfully completed.
+
+Some flows, such as login have additional query parameters which can be used in both browser and native applications.
+
+## Reusing an existing flow
+
+In cases where the flow already exists, we can retrieve the flow data using the flow ID. This is useful in cases where Ory has
+already initialized the flow and redirected to the UI. In such a case, the flow ID can be extracted from the URL `?flow=` query
+parameter.
+
+Here are some cases where the flow would already exist:
+
+1. Ory has initialized the flow and redirects to the UI (e.g. `/login?flow=<uuidv4>`)
+2. The request was completed by the user, however, an error occurred, and we need to retrieve the flow data to display the error
+3. The UI has 'stored' the flow ID in the URL so that page refreshes can retrieve the already existing flow data
+
+:::note
+
+The flow ID has an expiration time, so it is important to check the response status code when retrieving the flow data using an
+existing flow ID. It is up to your application to handle this error case and create a new flow.
+
+:::
+
+## Submitting a flow
+
+The flow always has to be initialized before any user data (e.g. email and password) can be submitted. Depending on the type of
+application (e.g. browser or native) the initialized flow will set a CSRF cookie and return a CSRF token in the response body. It
+is important in **browser based applications** to take note of the CSRF tokens since it is required when submitting the user data.
+
+:::warning
+
+Never use API flows to implement Browser applications! Using API flows in Single-Page-Apps as well as server-side apps opens up
+several potential attack vectors, including Login and other CSRF attacks.
+
+:::
+
+## Login flow
+
+Login flows are used to authenticate users. When the login is successful, the request will respond with a `200 OK` HTTP status
+code and a JSON body containing the user's identity information. When the request fails, it responds with the appropriate HTTP
+status and error messages.
```

Comment:
This is not really true. After successful login the server issues a session depending on the application (cookie or token).